integration_test/third_party_apps_test.go

```diff
@@ -379,6 +380,12 @@ func runSingleTest(ctx context.Context, logger *logging.DirectoryLogger, vm *gce
 		logger.ToMainLog().Printf("Parsed metadata.yaml: %+v", metadata)
 	}
 
+	// Validate the metadata.
+	validate := validator.New()
+	if err = validate.Struct(metadata); err != nil {
+		return nonRetryable, err
+	}
+
```

Comment:
I am also working on this :) It needs to do quite much to get it work in the we intended it to be.😮‍💨
metadata parsed by unmarshallstrict will allocate any substruct as {} instead of nill thus the required tag won't work. I will send a PR soon. For now you can restore this file back, i will handle the work in my PR :)

Reply:
Here is the PR: https://github.com/GoogleCloudPlatform/ops-agent/pull/567